app/src/main/java/com/example/candyshop/data/Resource.kt

```diff
@@ -0,0 +1,10 @@
+package com.example.candyshop.data
+
+sealed class Resource<T>(
```

Comment:
such wrapper for success/error scenario is great but on the level of data, so in repository I think you only need Success/Error scenario and loading state is more connected with ui layer so loading state i would create more like a ui state only - data layer should not be responsible for displaying loader. From data layer (repository) we are expecting to return some data (Success) or error (Error)